ratelimit/tokenbucket/tokenbucket_test.go

```diff
@@ -0,0 +1,22 @@
+package tokenbucket
```

Comment:
If we are getting the end user to pass in the rate limiter we can remove the whole tokenbucket package from this codebase and not depoend on juju ourselves.
The user would then wrap it using the Limiter interface so that we allow them to use any rate limit algorithm.

Reply:
I think the package can be kept in order to give user an easy implementation of `Limiter`.

And also user can implement his own limiter if it is not enough.

Reply:
we can give an example without needing to import the dependency, the thinking behind passing in the rate limiter is so that we do not have to maintain the rate limiters but just wrap them to meet the Limiter interface that has been defined.

Reply:
👌remove it from repo